JAVA-1968: Fix javadoc to show default value for a few methods

vedala opened the pull request:
Following modifications were made to javadoc information in file MongoClientOptions.java:

Added default value information:

```
 getMaxConnectionIdleTime
 getMaxConnectionLifeTime
```

Default value information moved to a separate line

```
getDbDecoderFactory
getDbEncoderFactory
```

Default value information moved to a separate line, modified slightly for consistency

```
getHeartbeatFrequency
getMinHeartbeatFrequency
getHeartbeatSocketTimeout
isSslEnabled
isSslInvalidHostNameAllowed
getCommandListeners
```

Default value information already present on separate line, modified slightly for consistency

```
getHeartbeatConnectTimeout
getLocalThreshold
```

Default value information modified from false to {@code false}

```
isAlwaysUseMBeans
isSocketKeepAlive                                 
```

rozza commented:
Hi @vedala,

Firstly, apologies for the lack of response on this pull request. We've been working hard on the new 3.4 release of the driver.  Please could you resync this to master and I'll look to merge?

All the best,

Ross

vedala commented:
Hi @rozza,

No problem about the delay. Glad you are considering merging this pull request. I have sync the master. But do I have to sync the feature branch to master too?

Thanks.

rozza commented:
Hi @vedala, thats correct as there currently is a conflict in the file preventing the merge.

vedala commented:
Hi @rozza, I have synced the feature branch to master. As you mentioned, there was a merge conflict which I resolved.

rozza commented on driver/src/main/com/mongodb/MongoClientOptions.java:
```diff
@@ -290,6 +290,8 @@ public int getMaxWaitTime() {
      * The maximum idle time of a pooled connection.  A zero value indicates no limit to the idle time.  A pooled connection that has
      * exceeded its idle time will be closed and replaced when necessary by a new connection.
      *
+     * <p>Default is 0.</p>
```
What does the default of 0 mean?

Change to `<p>Default is 0, indicating no limit to the idle time.</p>`

rozza replied:
What does the default of 0 mean?

Change to: `<p>Default is 0, indicating no limit to the idle time.</p>`

rozza commented on driver/src/main/com/mongodb/MongoClientOptions.java:
```diff
@@ -301,6 +303,8 @@ public int getMaxConnectionIdleTime() {
      * The maximum life time of a pooled connection.  A zero value indicates no limit to the life time.  A pooled connection that has
      * exceeded its life time will be closed and replaced when necessary by a new connection.
      *
+     * <p>Default is 0.</p>
```
Change to: `<p>Default is 0, indicating no limit to the life time.</p>`

rozza commented on driver/src/main/com/mongodb/MongoClientOptions.java:
```diff
@@ -346,7 +350,9 @@ public boolean isSocketKeepAlive() {
-     * cluster. The default value is 10,000 milliseconds.
+     * cluster.
+     *
+     * <p>Default is 10,000.</p>
```
It would be best to keep the timeunits for all the timeouts, so the user doesn't have to look it up. eg:

`<p>Default is 10,000.</p>`

rozza commented on driver/src/main/com/mongodb/MongoClientOptions.java:
```diff
@@ -547,7 +563,9 @@ public DBDecoderFactory getDbDecoderFactory() {
-     * Override the encoder factory. Default is for the standard Mongo Java driver configuration.
+     * Override the encoder factory.
+     *
+     * <p>Default is for the standard Mongo Java driver configuration.</p>
```
Lets add a default for `getDbDecoderFactory` as well.

vedala commented:
Hi @rozza, I have made the changes you suggested. I added back the timeunit for locations that I removed it from. I also added timeunit for few additional functions.

vedala commented:
There was a line too long checkStyle error on one line. I fixed it and pushed the changes.

rozza commented:
Many thanks @vedala
